docker_test.go

```diff
@@ -1187,6 +1189,115 @@ func TestEntrypoint(t *testing.T) {
 	defer c.Terminate(ctx)
 }
 
+func TestReadTCPropsFile(t *testing.T) {
+	t.Run("HOME does not contain TC props file", func(t *testing.T) {
+		oldHome := os.Getenv("HOME")
+		tmpDir := filet.TmpDir(t, "")
+		os.Setenv("HOME", tmpDir)
+		defer func() {
+			os.Setenv("HOME", oldHome)
+			filet.CleanUp(t)
+		}()
+
+		config := readTCPropsFile()
+
+		assert.Empty(t, config, "TC props file should not exist")
+	})
+
+	t.Run("HOME contains TC properties file", func(t *testing.T) {
+		oldHome := os.Getenv("HOME")
+
+		tests := []struct {
+			content           string
+			expectedHost      string
+			expectedTLSVerify int
+			expectedCertPath  string
+		}{
+			{
+				"docker.host = tcp://127.0.0.1:33293",
+				"tcp://127.0.0.1:33293",
+				0,
+				"",
+			},
+			{
+				"docker.host = tcp://127.0.0.1:33293",
+				"tcp://127.0.0.1:33293",
+				0,
+				"",
+			},
+			{
+				`docker.host = tcp://127.0.0.1:33293
+	docker.host = tcp://127.0.0.1:4711
+	`,
+				"tcp://127.0.0.1:4711",
+				0,
+				"",
+			},
+			{`docker.host = tcp://127.0.0.1:33293
+	docker.host = tcp://127.0.0.1:4711
+	docker.host = tcp://127.0.0.1:1234
+	docker.tls.verify = 1
+	`,
+				"tcp://127.0.0.1:1234",
+				1,
+				"",
+			},
+			{
+				"",
+				"",
+				0,
+				"",
+			},
+			{
+				`foo = bar
+	docker.host = tcp://127.0.0.1:1234
+			`,
+				"tcp://127.0.0.1:1234",
+				0,
+				"",
+			},
+			{
+				"docker.host=tcp://127.0.0.1:33293",
+				"tcp://127.0.0.1:33293",
+				0,
+				"",
+			},
+			{
+				`#docker.host=tcp://127.0.0.1:33293`,
+				"",
+				0,
+				"",
+			},
+			{
+				`#docker.host = tcp://127.0.0.1:33293
+	docker.host = tcp://127.0.0.1:4711
+	docker.host = tcp://127.0.0.1:1234
+	docker.cert.path=/tmp/certs`,
+				"tcp://127.0.0.1:1234",
+				0,
+				"/tmp/certs",
+			},
+		}
+		for _, tt := range tests {
+			tmpDir := filet.TmpDir(t, "")
+			os.Setenv("HOME", tmpDir)
+
+			defer func() {
+				os.Setenv("HOME", oldHome)
+				filet.CleanUp(t)
+			}()
+
+			_ = filet.File(t, path.Join(tmpDir, ".testcontainers.properties"), tt.content)
+
+			config := readTCPropsFile()
+
+			assert.Equal(t, tt.expectedHost, config.Host, "Hosts do not match")
```

Comment:
@kiview please take a look at the expected values for `Host` in the test struct, as reading the props file with `magiconair` will always set the value as the last entry in the props file.

Reply:
Right, that is actually the correct way how `properties` files should be read, so the hand-crafted implementation was flawed in this regard.